### PR TITLE
Bugfix KNOWAGE-8149

### PR DESCRIPTION
**Pull request by BojanSovticEngIT**
Bugfix [KNOWAGE-8149](https://production.eng.it/jira/browse/KNOWAGE-8149)

Files changed:

- knowage-api/src/main/java/it/eng/knowage/knowageapi/resource/FunctionCatalogResource.java
- knowage-core/src/main/java/it/eng/knowage/backendservices/rest/widgets/PythonProxy.java
- knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java
- knowage-core/src/main/java/it/eng/spagobi/api/BusinessModelResource.java
- knowage-core/src/main/java/it/eng/spagobi/api/common/AbstractDataSetResource.java
- knowage-core/src/main/java/it/eng/spagobi/api/v3/DataSetResource.java
- knowagecockpitengine/src/main/java/it/eng/knowage/engine/cockpit/api/page/PageResource.java
- knowageutils/src/main/java/it/eng/spagobi/utilities/engines/rest/SimpleRestClient.java

Possible false positives:

- RestExceptionMapper.java 
- KnowageBusinessExceptionMapper.java
- KnowageExceptionMapper.java
- KnowageRuntimeExceptionMapper.java  
- knowagecockpitengine/src/main/java/it/eng/knowage/engine/interceptors/RestExceptionMapper.java 
- knowageqbeengine/src/main/java/it/eng/spagobi/rest/interceptors/RestExceptionMapper.java
- knowagesvgviewerengine/src/main/java/it/eng/knowage/engines/svgviewer/interceptor/RestExceptionMapper.java  
- knowagewhatifengine/src/main/java/it/eng/spagobi/engines/whatif/exception/RestExceptionMapper.
java

### CODE REVIEW

File: knowageutils/src/main/java/it/eng/spagobi/utilities/engines/rest/SimpleRestClient.java
Comment (kerny3d) [inline knowageutils/src/main/java/it/eng/spagobi/utilities/engines/rest/SimpleRestClient.java:1]: I would say that this is a false positive too. The @GET, @POST and @PUT as all other HTTP methods annotations like PUT, PATH, DELETE, etc... don't make much sense when the class misses the @Path annotation. The SimpleRestClient is just an utility class to let BE makes REST call to other services: it's not its self a REST service. I would ask you to revert this class.

File: knowagecockpitengine/src/main/java/it/eng/knowage/engine/cockpit/api/page/PageResource.java
Comment (kerny3d) [inline knowagecockpitengine/src/main/java/it/eng/knowage/engine/cockpit/api/page/PageResource.java:1]: Nothing to say, just a check: is the Eclipse formatter correctly set up or it was an already unformatted class?

File: knowage-core/src/main/java/it/eng/spagobi/api/common/AbstractDataSetResource.java [hunk 1]
```diff
@@ -1048,7 +1048,7 @@ public String getDataSet(String label) {
 		}
 	}
 
-	public Response deleteDataset(String label) {
+	protected Response deleteDataset(String label) {
```
Comment (kerny3d) [inline knowage-core/src/main/java/it/eng/spagobi/api/common/AbstractDataSetResource.java:1051]: Do you think it's possible to use private? If there is no inheritance I would prefer the private modifier.

File: knowage-core/src/main/java/it/eng/spagobi/api/common/AbstractDataSetResource.java [hunk 2]
```diff
@@ -1121,7 +1121,7 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
 		}
 	}
 
-	public Response execute(String label, String body) {
+	protected Response execute(String label, String body) {
```
Comment (kerny3d) [inline knowage-core/src/main/java/it/eng/spagobi/api/common/AbstractDataSetResource.java:1124]: Do you think it's possible to use private? If there is no inheritance I would prefer the private modifier.

File: knowage-api/src/main/java/it/eng/knowage/knowageapi/resource/FunctionCatalogResource.java [hunk 3]
```diff
@@ -138,7 +139,7 @@ public Response delete(@PathParam("id") UUID id) {
 		}
 	}
 
-	@PATCH
+	@PUT
```
Comment (kerny3d) [inline knowage-api/src/main/java/it/eng/knowage/knowageapi/resource/FunctionCatalogResource.java:142]: Is it bug fixing? :-D Just kidding, great job.

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java [hunk 4]
```diff
@@ -50,7 +50,7 @@ public abstract class AbstractDocumentResource extends AbstractSpagoBIResource {
 
 	private static final Logger LOGGER = LogManager.getLogger(AbstractDocumentResource.class);
 
-	public Response insertDocument(String body) {
+	protected Response insertDocument(String body) {
```
Comment (kerny3d) [inline knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:53]: If possible, I would prefer private but this is an abstract class, we need a check.

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java [hunk 5]
```diff
@@ -88,7 +88,7 @@ public Response insertDocument(String body) {
 		}
 	}
 
-	public Response getDocumentDetails(Object documentIdentifier) {
+	protected Response getDocumentDetails(Object documentIdentifier) {
```
Comment (kerny3d) [inline knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:91]: If possible, I would prefer private but this is an abstract class, we need a check.

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java [hunk 6]
```diff
@@ -115,7 +115,7 @@ public Response getDocumentDetails(Object documentIdentifier) {
 
 	}
 
-	public Response updateDocument(String label, String body) {
+	protected Response updateDocument(String label, String body) {
```
Comment (kerny3d) [inline knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:118]: If possible, I would prefer private but this is an abstract class, we need a check.

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java [hunk 7]
```diff
@@ -133,7 +133,7 @@ public Response updateDocument(String label, String body) {
 		return Response.ok().build();
 	}
 
-	public Response deleteDocument(@PathParam("label") String label) {
+	protected Response deleteDocument(@PathParam("label") String label) {
```
Comment (kerny3d) [inline knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:136]: If possible, I would prefer private but this is an abstract class, we need a check.

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java [hunk 8]
```diff
@@ -167,7 +167,7 @@ protected Object getObjectIdentifier(String labelOrId) {
 		return documentIdentifier;
 	}
 
-	public Response getDocumentTemplate(String label) {
+	protected Response getDocumentTemplate(String label) {
```
Comment (kerny3d) [inline knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:170]: If possible, I would prefer private but this is an abstract class, we need a check.

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java [hunk 9]
```diff
@@ -194,7 +194,7 @@ public Response getDocumentTemplate(String label) {
 		return rb.build();
 	}
 
-	public Response addDocumentTemplate(String label, MultiPartBody input) {
+	protected Response addDocumentTemplate(String label, MultiPartBody input) {
```
Comment (kerny3d) [inline knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:197]: If possible, I would prefer private but this is an abstract class, we need a check.

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java [hunk 10]
```diff
@@ -229,7 +229,7 @@ public Response addDocumentTemplate(String label, MultiPartBody input) {
 
 	}
 
-	public Response deleteCurrentTemplate(String documentLabel) {
+	protected Response deleteCurrentTemplate(String documentLabel) {
```
Comment (kerny3d) [inline knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:232]: If possible, I would prefer private but this is an abstract class, we need a check.

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java [hunk 11]
```diff
@@ -257,7 +257,7 @@ public Response deleteCurrentTemplate(String documentLabel) {
 		return Response.ok().build();
 	}
 
-	public Response deleteTemplateById(String documentLabel, Integer templateId) {
+	protected Response deleteTemplateById(String documentLabel, Integer templateId) {
```
Comment (kerny3d) [inline knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:260]: If possible, I would prefer private but this is an abstract class, we need a check.

**Review by kerny3d — COMMENTED** [review]
Need some checks

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java (hunk 11 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:260]: This one can't be private, it is being used in DocumentResource.java.

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java (hunk 10 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:232]: This one can't be private, it is being used in DocumentResource.java.

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java (hunk 9 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:197]: This one can't be private, it is being used in DocumentResource.java (two files, one of them is v2).

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java (hunk 8 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:170]: This one can't be private, it is being used in DocumentResource.java (two files, one of them is v2).

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java (hunk 7 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:136]: This one can't be private, it is being used in DocumentResource.java (two files, one of them is v2).

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java (hunk 6 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:118]: This one can't be private, it is being used in DocumentResource.java (two files, one of them is v2).

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java (hunk 5 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:91]: This one can't be private, it is being used in DocumentResource.java (two files, one of them is v2).

File: knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java (hunk 4 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/api/AbstractDocumentResource.java:53]: This one can't be private, it is being used in DocumentResource.java (two files, one of them is v2).

File: knowage-core/src/main/java/it/eng/spagobi/api/common/AbstractDataSetResource.java (hunk 2 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/api/common/AbstractDataSetResource.java:1124]: This one can't be private, it is being used in DataSetResource.java (two files, one of them is v2).

File: knowage-core/src/main/java/it/eng/spagobi/api/common/AbstractDataSetResource.java (hunk 1 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/api/common/AbstractDataSetResource.java:1051]: This one can't be private, it is being used in DataSetResource.java (two files, one of them is v2).

File: knowagecockpitengine/src/main/java/it/eng/knowage/engine/cockpit/api/page/PageResource.java
Comment (BojanSovticEngIT) [reply knowagecockpitengine/src/main/java/it/eng/knowage/engine/cockpit/api/page/PageResource.java:1]: We set the Eclipse formatting with Davide Z., it should be exactly the same as you use it, but a lot of the (older) files are unformatted. This is a common issue when creating pull requests and looking for changes.

File: knowageutils/src/main/java/it/eng/spagobi/utilities/engines/rest/SimpleRestClient.java
Comment (BojanSovticEngIT) [reply knowageutils/src/main/java/it/eng/spagobi/utilities/engines/rest/SimpleRestClient.java:1]: Removed @get, @post and @put

**Pull request merged**